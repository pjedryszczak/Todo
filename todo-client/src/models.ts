export interface Todo {
        id?: number,
        content: string,
        todoListId: number,
        checked: boolean
    }
export interface TodoList {
        id?: number,
        userId: number,
        title: string
    }   

 