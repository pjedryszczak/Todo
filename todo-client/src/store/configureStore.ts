import {createStore, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";
import createSagaMiddleware from "redux-saga";
import sagas from "./sagas";
import { reducer } from "./reducer";

export default function configureStore(initialState: any, history: any) {
    const composeEnhancers = composeWithDevTools({});
    const reduxSagaMiddleware = createSagaMiddleware();

    const middleware = [reduxSagaMiddleware, thunk ];
    const applyMiddlewareFunc = applyMiddleware(...middleware);

    let store: any;
    if(process.env.NODE_ENV === 'production') {
        store = createStore(reducer, applyMiddlewareFunc);
    }else{
        store = createStore(reducer,initialState, composeEnhancers(applyMiddlewareFunc));
    }

    reduxSagaMiddleware.run(sagas);

    return store;
}