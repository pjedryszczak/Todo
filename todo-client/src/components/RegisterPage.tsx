import React, { ChangeEvent, FormEvent } from 'react';
import { Link, RouteComponentProps } from 'react-router-dom';
import { connect } from 'react-redux';
import { register } from './../store/actions';
import { User, RegisterPayload, RegisterModel } from '../models';
import { TodoAppState } from '../store/reducer';


interface DispatchProps {
    register: typeof register
}
interface StoreState {
    user: User,
    loggedIn: boolean,
    loading: boolean
}

interface LocalState {
    [key: string]: any;
    username: string,
    password: string,
    firstName: string,
    submitted: boolean
}

type Props = DispatchProps & StoreState & LocalState & RouteComponentProps;
class LoginPage extends React.Component<Props, LocalState> {
    constructor(props:any) {
        super(props);

        this.state = {
            username: '',
            password: '',
            firstName: '',
            submitted: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(e: ChangeEvent<HTMLInputElement>) {
        const { name, value } = e.target;
        this.setState({ [name]: value });
    }

    handleSubmit(e: FormEvent<HTMLFormElement>) {
        e.preventDefault();

        this.setState({ submitted: true });
        const { username, password, firstName } = this.state;
        const {history} = this.props;
        if (username && password) {
            const registerModel: RegisterModel ={
                username,
                password,
                firstName,
            }
            const payload: RegisterPayload ={
                payload: registerModel,
                history
            }
            this.props.register(payload);
        }
    }

    render() {
        const { loading } = this.props;
        const { username, password, firstName, submitted } = this.state;
        return (
            <div>
                <h2>Register</h2>
                <form name="form" onSubmit={this.handleSubmit}>
                    <div className={'form-group' + (submitted && !firstName ? ' has-error' : '')}>
                        <label htmlFor="firstName">First Name</label>
                        <input type="text" className="form-control" name="firstName" value={firstName} onChange={this.handleChange} />
                        {submitted && !firstName &&
                        <span data-error="wrong" >First Name is required</span>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !username ? ' has-error' : '')}>
                        <label htmlFor="username">Username</label>
                        <input type="text" className="form-control" name="username" value={username} onChange={this.handleChange} />
                        {submitted && !username &&
                        <span data-error="wrong" >Username is required</span>
                        }
                    </div>
                    <div className={'form-group' + (submitted && !password ? ' has-error' : '')}>
                        <label htmlFor="password">Password</label>
                        <input type="password" className="form-control" name="password" value={password} onChange={this.handleChange} />
                        {submitted && !password &&
                            <span data-error="wrong" style={{textDecorationColor: 'red'}} >Password is required</span>
                        }
                    </div>
                    <div className="form-group">
                        <button className="waves-effect waves-light blue darken-1 btn">Register</button>
                        {loading &&
                            <>≈ÅADOWANIE</>
                        }
                        <Link to="/" className="waves-effect waves-light green darken-1 btn">Cancel</Link>
                    </div>
                </form>
            </div>
        );
    }
}
function mapStateToProps(state: TodoAppState, props: any){
    return {
        loading: state.loading
    }
}
const container = connect(
    mapStateToProps,
    {
    register
}
)(LoginPage)

export default container;