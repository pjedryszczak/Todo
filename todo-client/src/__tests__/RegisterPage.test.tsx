import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import RegisterPage from '../components/RegisterPage';
import { Provider } from 'react-redux';
import { User } from '../models';
import { StaticRouter } from 'react-router-dom';

const mockStore = configureStore([]);
describe('LoginPage', () => {
    let store: any;
    let component: renderer.ReactTestRenderer; 

   const user: User ={
    firstName: 'TestName',
    id: 0,
    username: 'testname',
    token: ''
   }
   const event = {
       preventDefault: () => {}
   } as React.FormEvent<HTMLFormElement>;
    beforeEach(() => {
      store = mockStore({
        loggedIn: false,
        user: user,
        loading: false
      });
      store.dispatch = jest.fn();

      component = renderer.create(
          <StaticRouter>
        <Provider store={store}>
            <RegisterPage />
        </Provider>
        </StaticRouter>);
    });

    it('it renders with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
    it('register action on submit', () => {
        
        renderer.act(() => {
            component.root.findByType('form').props.onSubmit(event);
          });
          //inputs are empty
        expect(store.dispatch).toHaveBeenCalledTimes(0);        
      });
  });