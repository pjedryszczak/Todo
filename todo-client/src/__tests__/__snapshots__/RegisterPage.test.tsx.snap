// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage it renders with given state from Redux store 1`] = `
<div>
  <h2>
    Register
  </h2>
  <form
    name="form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="firstName"
      >
        First Name
      </label>
      <input
        className="form-control"
        name="firstName"
        onChange={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="waves-effect waves-light blue darken-1 btn"
      >
        Register
      </button>
      <a
        className="waves-effect waves-light green darken-1 btn"
        href="/"
        onClick={[Function]}
      >
        Cancel
      </a>
    </div>
  </form>
</div>
`;
